src/lib/cache/lru_k_cache.hpp

```diff
@@ -57,6 +57,34 @@ class LRUKCache : public AbstractCacheImpl<Key, Value> {
   typedef LRUKCacheEntry entry_t;
   typedef typename boost::heap::fibonacci_heap<entry_t>::handle_type handle_t;
```

Comment:
Not your code, but this violates the code style twice. First, because of the naming, second, because we don't use `typedef` but `using`.  Could you have a look at that and check if other files in the cache folder do the same?